### PR TITLE
Live Site

### PR DESCRIPTION
**Pull request by adriandarian**
# Proposed changes

Live site updates

## Types of changes

What types of changes does your code introduce to HackMerced Hub?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/HackMerced/HackMerced/blob/master/CONTRIBUTING.md) doc
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Responsiveness

Check off the different **browsers** and **devices** you have tested on.

### Browsers

- [x] Chrome
- [ ] Firefox
- [ ] Edge
- [ ] Safari

### Devices

- [x] Windows
- [x] MacOSX
- [ ] Linux
- [x] IPhone
- [x] Android
- [x] Tablet

## Screenshots

N/A

## Further comments

N/A

### CODE REVIEW

File: .prettierrc.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module.exports = {
     semi: true,
     trailingComma: 'all',
-    singleQuote: true,
+    singleQuote: false,
```
Comment (4shub) [inline .prettierrc.js:4]: why

File: package.json [hunk 2]
```diff
@@ -1,17 +1,20 @@
...
         "predeploy": "yarn build",
         "deploy": "gh-pages -d build -b gh-pages",
         "gren": "gren release --override --tags=all --config=.grenrc.js",
+        "lint": "eslint --ext .ts,.tsx .",
+        "lint:fix": "yarn lint --fix",
+        "prestart ": "yarn lint ",
         "start": "yarn tailwind:css && react-scripts start",
         "storybook": "start-storybook -p 9001 -s ./src/assets -c .storybook",
-        "tailwind:css": "tailwind build src/tailwind.css -c tailwind.config.js -o src/tailwind.scss",
+        "tailwind:css": "npx tailwindcss-cli@latest build src/tailwind.css -c tailwind.config.js -o src/tailwind.scss",
```
Comment (4shub) [inline package.json:17]: You should rely on local dependencies for processing files, not latest cli

File: src/App.types.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    allergies: string;
+    numberOfPreviousHackathons: number;
+    resume: string;
+    linkedIn: string;
+    gitHub: string;
+    devpost: string;
+    otherSites: string;
+    teamCode: string;
+    teamName: string;
+    teammates: [string];
```
Comment (4shub) [inline src/App.types.ts:24]: `string[]` is array of strings, [string] means an array with a single element, which is string

File: src/components/Accordian/index.tsx [hunk 4]
```diff
@@ -1,18 +1,18 @@
...
 
 const Accordian: FC<AccordianProps> = ({ data }) => {
     const [accordionItems, setAccodionItems] = useState<AccordianState>([]);
 
     useEffect(() => {
         const parseData: () => void = (): void => {
-            let accordion: AccordianState = [];
+            const accordion: AccordianState = [];
 
             data.forEach((i: { title: string; content: string }) => {
```
Comment (4shub) [inline src/components/Accordian/index.tsx:17]: why not map instead of foreach

File: src/components/HackMercedApplication/index.tsx [hunk 5]
```diff
@@ -199,8 +203,8 @@ const Application: FC = (): JSX.Element => {
                             <label htmlFor="gender">
                                 Gender <span className="required">*</span>
                             </label>
-                            <select id="gender" name="gender" onChange={handleInputChange} required>
-                                <option value="" disabled={true} selected={true}>
+                            <select id="gender" name="gender" onChange={handleInputChange} required defaultValue="">
+                                <option value="" disabled={true}>
                                     Gender
                                 </option>
                                 <option value="male">Male</option>
```
Comment (4shub) [inline src/components/HackMercedApplication/index.tsx:210]: Use non-binary over other, it's more inclusive and less othering (literally)

File: src/components/NavBar/index.tsx [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+    textColor,
+    breakLineColor,
+    showBanner = true,
+}): JSX.Element => {
+    let { pathname } = useLocation();
+    const [openDrawer, toggleDrawer] = useState(false);
+    const drawerRef = useRef(null);
+
+    useEffect(() => {
+        const closeDrawer = (event: { target: any }) => {
```
Comment (4shub) [inline src/components/NavBar/index.tsx:19]: `MouseEvent` btw

File: src/components/NavBar/index.tsx [hunk 7]
```diff
@@ -0,0 +1,223 @@
...
+                    ) : (
+                        <div className="hmtext">HackMerced</div>
+                    )}
+                </Menu.Logo>
+
+                <HamburgerButton.Wrapper onClick={() => toggleDrawer(true)}>
+                    <HamburgerButton.Lines breakLineColor={breakLineColor} />
+                </HamburgerButton.Wrapper>
+
+                <Menu.Items ref={drawerRef} {...NavbarItemsProps} style={{ backgroundColor }}>
```
Comment (4shub) [inline src/components/NavBar/index.tsx:54]: does onMouseEnter/onMouseLeave not work here?

File: src/components/TeamPics/index.tsx [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+import MARTHEN from "../../assets/images/marty.jpg";
+import LAURENCE from "../../assets/images/laurence.png";
+import JET from "../../assets/images/jet.jpg";
+import JOSEPH from "../../assets/images/joseph.jpg";
+import JANELLE from "../../assets/images/janelle.jpg";
+
+import "./styles.scss";
+
+const TeamPics: FC = (): JSX.Element => {
+    const team: Array<{ url: string; src: string }> = [
```
Comment (4shub) [inline src/components/TeamPics/index.tsx:23]: Supernit: You should probably declare this outside the component, as it causes an additional memory allocation event even though the data is not changing

File: src/components/TeamPics/index.tsx [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+        {
+            url: "https://www.linkedin.com/in/jpjanelleperez/",
+            src: JANELLE,
+        },
+    ];
+
+    const teampics = (team: Array<{ url: string; src: string }>): JSX.Element => {
+        const teamArray: Array<JSX.Element> = [];
+
+        team.forEach((member: { url: string; src: string }, index: number) => {
```
Comment (4shub) [inline src/components/TeamPics/index.tsx:93]:
```
const teamArray: Array<JSX.Element> = team.map((member: { url: string; src: string }, index: number) => {
    const name: string = member.src.includes("/")
                ? member.src.split("/")[3].split(".")[0]
                : member.src.split(".")[0];
    return (
                <li key={index}>
                    <div className="hexagon">
                        <a href={member.url} target="_blank" rel="noopener noreferrer">
                            <img
                                className="Picture"
                                src={member.src}
                                alt={name.charAt(0).toUpperCase() + name.slice(1)}
                            />
                        </a>
                    </div>
                </li>,
            );
})
```

Always prefer map over forEach+push

File: src/components/WindowDimensions/index.tsx [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+        height,
+    };
+}
+
+export default function useWindowDimensions(): {
+    width: number;
+    height: number;
+} {
+    const [windowDimensions, setWindowDimensions] = useState<{
+        width: number;
```
Comment (4shub) [inline src/components/WindowDimensions/index.tsx:16]: Simplify and make 
```
{
    width: number;
    height: number;
}
```

Into it's own type

File: src/pages/ResetPassword/index.tsx [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+        const { name, value } = event.target;
+        setForm({ ...form, [name]: value });
+    };
+
+    const handleSubmit = (event: React.ChangeEvent<HTMLFormElement>): void => {
+        event.preventDefault();
+        Axios.post(
+            `https://hackmerced-tomoe.herokuapp.com/v1/auth/login-resetpassword`,
+            form,
+        ).then((response: AxiosResponse) => console.log(response));
```
Comment (4shub) [inline src/pages/ResetPassword/index.tsx:24]: Shouldn't you give a response to this?

**Review by 4shub — CHANGES_REQUESTED** [review]
I would leave more but I got tired lol but looks good generally

File: .prettierrc.js (hunk 1 above)
Comment (adriandarian) [reply .prettierrc.js:4]: I like single quotes tho, but ok

File: package.json (hunk 2 above)
Comment (adriandarian) [reply package.json:17]: we are gonna remove tailwind soon so this will be obsolete

File: src/App.types.ts (hunk 3 above)
Comment (adriandarian) [reply src/App.types.ts:24]: this is from old code that is not gonna be used in production

File: src/components/Accordian/index.tsx (hunk 4 above)
Comment (adriandarian) [reply src/components/Accordian/index.tsx:17]: cause I have not practiced using the array prototypes as much and been sticking to base stuff

File: src/pages/ResetPassword/index.tsx (hunk 11 above)
Comment (adriandarian) [reply src/pages/ResetPassword/index.tsx:24]: not needed anymore

File: src/components/TeamPics/index.tsx (hunk 8 above)
Comment (adriandarian) [reply src/components/TeamPics/index.tsx:23]: did not know tyty

File: src/components/NavBar/index.tsx (hunk 6 above)
Comment (adriandarian) [reply src/components/NavBar/index.tsx:19]: tyty, but I feel like there was a reason why I did not put it

File: src/components/HackMercedApplication/index.tsx (hunk 5 above)
Comment (adriandarian) [reply src/components/HackMercedApplication/index.tsx:210]: ok

**Pull request merged**